src/suricata-plugin.h

```diff
@@ -0,0 +1,30 @@
+#ifndef __SURICATA_PLUGIN_H__
+#define __SURICATA_PLUGIN_H__
+
+#define SURICATA_PLUGIN 1
+
+#include <config.h>
+
+#undef HAVE_LUA
+#undef HAVE_NSS
+
+#include "suricata-common.h"
+#include "util-logopenfile.h"
+
+typedef struct SCPlugin_ {
+    char *name;
+    char *license;
+    char *author;
+
+    void (*Init)(void);
+} SCPlugin;
+
+typedef struct PluginFileType_ {
+    char *name;
+    int (*Open)(LogFileCtx *ctx, ConfNode *conf);
```

Comment:
I agree this should not expose `LogFileCtx` for compatibility reasons.

Reply:
My current working branch has removed exposing LogFileCtx. Resolving this comment.